crates/services/gas_price_service/src/v1/metadata.rs

```diff
@@ -0,0 +1,142 @@
+use crate::v0::metadata::V0Metadata;
+use fuel_gas_price_algorithm::v1::{
+    AlgorithmUpdaterV1,
+    ClampedPercentage,
+    L2ActivityTracker,
+};
+use std::num::NonZeroU64;
+
+#[derive(serde::Serialize, serde::Deserialize, Debug, Clone, PartialEq)]
+pub struct V1Metadata {
+    // Execution
+    /// The gas price (scaled by the `gas_price_factor`) to cover the execution of the next block
+    pub new_scaled_exec_price: u64,
+    /// The height of the next L2 block
+    pub l2_block_height: u32,
+    // DA
+    /// The gas price (scaled by the `gas_price_factor`) to cover the DA commitment of the next block
+    pub new_scaled_da_gas_price: u64,
+    /// Scale factor for the gas price.
+    pub gas_price_factor: NonZeroU64,
+    /// The cumulative reward from the DA portion of the gas price
+    pub total_da_rewards_excess: u128,
+    /// The height of the last L2 block recorded on the DA chain
+    pub da_recorded_block_height: u32,
+    /// The cumulative cost of recording L2 blocks on the DA chain as of the last recorded block
+    pub latest_known_total_da_cost_excess: u128,
+    /// The predicted cost of recording L2 blocks on the DA chain as of the last L2 block
+    /// (This value is added on top of the `latest_known_total_da_cost` if the L2 height is higher)
+    pub projected_total_da_cost: u128,
+    /// The last profit
+    pub last_profit: i128,
+    /// The profit before last
+    pub second_to_last_profit: i128,
+    /// The latest known cost per byte for recording blocks on the DA chain
+    pub latest_da_cost_per_byte: u128,
+    /// The l2 blocks that have not been recorded on the DA chain, but have been used to estimate
+    /// the cost of recording blocks on the DA chain
+    pub unrecorded_blocks: Vec<(u32, u64)>,
+}
+
+impl V1Metadata {
+    pub fn construct_from_v0_metadata(
+        v0_metadata: V0Metadata,
+        config: V1AlgorithmConfig,
+    ) -> anyhow::Result<Self> {
+        let gas_price_factor = NonZeroU64::new(config.gas_price_factor)
```

Comment:
Is it possible to directly have a `NonZeroU64` in the config ?

Reply:
Sure. but we need to convert it someplace. I don't think it's possible using `clap`. I don't have a good set of rules around where that conversion should happen.

Reply:
We are using it with clap here : https://github.com/FuelLabs/fuel-core/blob/328b42ca5d26f7ca74b979df38137e001b5a86ed/bin/fuel-core/src/cli/run/p2p.rs#L179 

I think the sooner the error is handled the better it is. To avoid anyone using the value before and assuming 0 is possible

Reply:
Yep. I'll change it. Thanks!